features/request_construction/headers.feature

```diff
@@ -1,60 +1,81 @@
 Feature: Adding Headers
 
-  Backround:
-    Given brine is mixed
-
   Scenario: Headers are added to requests.
     Given a file named "features/headers.feature" with:
 
       """
 Feature: Request headers can be specified.
   Scenario: A new header with a single value is added to request.
+    Given expected request headers:
+      \"\"\"
+      {"foo":"bar"}
+      \"\"\"
+    And expected GET sent to `/query`
+
     When the request header `foo` is assigned `bar`
     And a GET is sent to `/query`
 
-    Then there was a GET request with a url matching `/query`
-    And it had headers including `{"foo":"bar"}`
-    And it was sent
+    Then expected calls are verified
 
   Scenario: Default headers are present in requests.
+    Given expected request headers:
+      \"\"\"
+      {"Content-Type": "application/json"}
+      \"\"\"
+    And expected GET sent to `/query`
+
     When a GET is sent to `/query`
 
-    Then there was a GET request with a url matching `/query`
-    And it had headers including `{"Content-Type": "application/json"}`
-    And it was sent
+    Then expected calls are verified
 
   Scenario: Default headers can be overridden.
+    Given expected request headers:
+      \"\"\"
+      {"Content-Type": "text/plain"}
```

Comment:
these changes have nothing to do with resource cleanup, right?

Reply:
These are some of the changes from #90, they'll disappear after rebasing before merge